pkg/controller/deployment/util/deployment_util.go

```diff
@@ -582,7 +582,12 @@ func LabelPodsWithHash(podList *api.PodList, rs *extensions.ReplicaSet, c client
 	for _, pod := range podList.Items {
 		// Only label the pod that doesn't already have the new hash
 		if pod.Labels[extensions.DefaultDeploymentUniqueLabelKey] != hash {
-			if _, podUpdated, err := podutil.UpdatePodWithRetries(c.Core().Pods(namespace), &pod,
+			copiedPod, err := PodDeepCopy(&pod)
```

Comment:
No need to deep-copy here since `UpdatePodWithRetries` does a live GET below. We will need it once we get rid of the api call: https://github.com/kubernetes/kubernetes/issues/34534